cmake/kokkos_corner_cases.cmake

```diff
@@ -46,15 +46,3 @@ IF(KOKKOS_CXX_COMPILER_ID STREQUAL AppleClang AND KOKKOS_ENABLE_OPENMP)
   UNSET(APPLECLANG_HAS_OMP CACHE) #make sure CMake always re-runs this
 ENDIF()
 
-
-IF (KOKKOS_CXX_COMPILER_ID STREQUAL GNU AND KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 7)
-  MESSAGE(FATAL_ERROR "The GCC version used is ${KOKKOS_CXX_COMPILER_VERSION} but GCC < 7 does not properly support *this capture. Please upgrade the compiler.")
-ENDIF()
-
-IF (KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA AND KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 11)
-  MESSAGE(FATAL_ERROR "Kokkos requires C++17 which NVCC only supports from version 11 on but the NVCC version used is ${KOKKOS_CXX_COMPILER_VERSION}.")
-ENDIF()
-IF (KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA AND KOKKOS_ENABLE_CUDA_CONSTEXPR AND KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 11.2)
-  MESSAGE(WARNING "You have requested -DKokkos_ENABLE_CUDA_CONSTEXPR=ON for NVCC ${KOKKOS_CXX_COMPILER_VERSION} which is known to trigger compiler bugs before NVCC version 11.2. See https://github.com/kokkos/kokkos/issues/3496")
-ENDIF()
```

Comment:
We haven't set nvcc 11.2 as a requirement, so I think this needs to stay

Reply:
yes